ADMVEDU/ViewControllers/ViewController.swift

```diff
@@ -42,49 +63,54 @@ class ViewController: UIViewController {
         tableView.delegate = self
         tableView.dataSource = self
     }
+
 }
 
 // MARK: Configure SearchBar
 extension ViewController: UISearchBarDelegate {
-    
+
     func searchBarSearchButtonClicked(_ searchBar: UISearchBar) {
         if let text = searchBar.text {
             requestOptions.term = text
             fetchMedia(options: requestOptions)
         }
     }
-    
+
+    func searchBarTextDidEndEditing(_ searchBar: UISearchBar) {
+        searchController.isActive = false
+        searchBar.text = requestOptions.term
+    }
+
 }
 
 // MARK: Configure TableView
 extension ViewController: UITableViewDelegate, UITableViewDataSource {
     func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
         return results.count
     }
-    
+
     func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
-        let cell = tableView.dequeueReusableCell(withIdentifier: MediaCell.identifier) as! MediaCell
+        guard let cell = tableView.dequeueReusableCell(withIdentifier: MediaCell.identifier) as? MediaCell else {
+            return MediaCell()
+        }
```

Comment:
To get rid of code duplication just use as? operator in first line and then "guard let cell = cell..."